src/vsphere_cpi/lib/cloud/vsphere/vm_creator.rb

```diff
@@ -208,13 +247,24 @@ def create_drs_rules(vm_config, vm_mob, cluster)
       drs_rule.add_vm(vm_mob)
     end
 
-    def add_vm_to_vm_group(vm_config, vm_mob, cluster)
-      return if vm_config.vm_type.vm_group.nil?
-      vm_group = VSphereCloud::VmGroup.new(
+    def add_vm_to_vm_group(vm_group, vm_mob, cluster)
+      return if vm_group.nil?
+      vm_group_creator = VSphereCloud::VmGroup.new(
+        @client,
+        cluster.mob
+      )
+      vm_group_creator.add_vm_to_vm_group(vm_mob, vm_group)
+    end
+
+    def create_vm_host_affinity_rule(vm_mob, cluster)
+      return if cluster.host_group.nil?
```

Comment:
This check is already happening before the method is being called, either you can remove this, or remove the check when this method is called as, add_vm_to_vm_group method also has a check if vm_group.nil?

Reply:
I had a TODO here before with title `should we make a function secure or the callee place should make sure of this` 

Since this is a local cluster resource and check will happen very fast, I decided to keep it here also in case if this function is used in future.
What do you say?